src/msgraph_core/requests/batch_response_content.py

```diff
@@ -0,0 +1,134 @@
+from typing import Optional, Dict, List, Type, TypeVar, Callable
+from io import BytesIO
+import base64
+
+from kiota_abstractions.serialization import Parsable
+from kiota_abstractions.serialization import ParseNode
+from kiota_abstractions.serialization import ParseNodeFactoryRegistry
+from kiota_abstractions.serialization import SerializationWriter
+
+from .batch_response_item import BatchResponseItem
+
+T = TypeVar('T', bound='Parsable')
+
+
+class BatchResponseContent(Parsable):
+
+    def __init__(self) -> None:
+        """
+        Initializes a new instance of the BatchResponseContent class.
+        BatchResponseContent is a collection of BatchResponseItem items,
+         each with a unique request ID.
+        """
+        self._responses: Optional[List['BatchResponseItem']] = []
+
+    @property
+    def responses(self) -> Optional[List['BatchResponseItem']]:
+        """
+        Get the responses in the collection
+        :return: A dictionary of response IDs and their BatchResponseItem objects
+        :rtype: Optional[Dict[str, BatchResponseItem]]
+        """
+        return self._responses
+
+    @responses.setter
+    def responses(self, responses: Optional[List['BatchResponseItem']]) -> None:
+        """
+        Set the responses in the collection
+        :param responses: The responses to set in the collection
+        :type responses: Optional[Dict[str, BatchResponseItem]]
+        """
+        if isinstance(responses, dict):
+            self._responses = {response.id: response for response in responses.values()}
+        else:
+            self._responses = responses
+
+    def response(self, request_id: str) -> 'BatchResponseItem':
+        """
+        Get a response by its request ID from the collection
+        :param request_id: The request ID of the response to get
+        :type request_id: str
+        :return: The response with the specified request ID as a BatchResponseItem
+        :rtype: BatchResponseItem
+        """
+        if self._responses is None:
+            raise ValueError("Responses list is not initialized.")
+        for response in self._responses:
+            if response.request_id == request_id:
+                return response
+        raise KeyError(f"Response with request ID {request_id} not found.")
+
+    def get_response_by_id(self, request_id: str) -> 'BatchResponseItem':
+        if self._responses is None:
+            raise ValueError("Responses list is not initialized.")
+        for response in self._responses:
+            if response.request_id == request_id:
+                return response
+        raise KeyError(f"Response with request ID {request_id} not found.")
+
+    def response_body(self, request_id: str, type: Type[T]) -> Optional[T]:
```

Comment:
I think we may be missing a function for getting back the response as raw stream content. e.g. When a user uses batch to fetch the photo or attachment contents using batching.  
https://github.com/microsoftgraph/msgraph-sdk-design/blob/main/content/BatchResponseContent.md#usage-examples

Reply:
Done, updates samples with:

```python
try:
        stream_response = batch_response_content.get_response_stream_by_id(batch_request_item1.id)
        print(f"Stream Response: {stream_response}")
        print(f"Stream Response Content: {stream_response.read()}")
    except AttributeError as e:
        print(f"Error getting response by ID: {e}")
```